api/fees/data.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2025 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package fees
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/vechain/thor/v2/api/utils"
+	"github.com/vechain/thor/v2/bft"
+	"github.com/vechain/thor/v2/block"
+	"github.com/vechain/thor/v2/cache"
+	"github.com/vechain/thor/v2/chain"
+	"github.com/vechain/thor/v2/thor"
+)
+
+func newFeesData(repo *chain.Repository, bft bft.Committer, backtraceLimit uint32, fixedSize uint32) *FeesData {
+	return &FeesData{
+		repo:           repo,
+		cache:          cache.NewPrioCache(int(fixedSize)),
+		bft:            bft,
+		backtraceLimit: backtraceLimit,
+	}
+}
+
+func getBaseFee(baseFee *big.Int) *hexutil.Big {
+	if baseFee != nil {
+		return (*hexutil.Big)(baseFee)
+	}
+	return (*hexutil.Big)(big.NewInt(0))
+}
+
+func (fd *FeesData) pushToCache(header *block.Header) {
+	feeCacheEntry := &FeeCacheEntry{
+		baseFee:       getBaseFee(header.BaseFee()),
+		gasUsedRatio:  float64(header.GasUsed()) / float64(header.GasLimit()),
+		parentBlockID: header.ParentID(),
+	}
+	fd.cache.Set(header.ID(), feeCacheEntry, float64(header.Number()))
+}
+
+func (fd *FeesData) resolveRange(newestBlockSummary *chain.BlockSummary, blockCount uint32) (*utils.Revision, []*hexutil.Big, []float64, error) {
```

Comment:
Worth adding a comment that it's expected that the ranges of newestBlockSummary - blockCount are already sanitized.

Reply:
Not sure why that comment was lost but I think it's ok to replace
```
newestBlockID, err := fd.repo.NewBestChain().GetBlockID(newestBlockSummary.Header.Number())
	if err != nil {
		return thor.Bytes32{}, nil, nil, err
```

with 
`newestBlockID := newestBlockSummary.Header.ID()`